parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.kt

```diff
@@ -24,76 +25,94 @@ import android.content.Context
-    )
-    public fun trackPause()
-    public fun resetVideo()
-    public fun flushEventQueue()
-    public fun flushTimerIsActive(): Boolean
-
-    public companion object {
-        private const val DEFAULT_FLUSH_INTERVAL_SECS = 60
-        private var instance: ParselyTrackerInternal? = null
-
-        /**
-         * Singleton instance accessor. Note: This must be called after [.sharedInstance]
-         *
-         * @return The singleton instance
-         */
-        @JvmStatic
-        public fun sharedInstance(): ParselyTracker? {
-            return instance
-        }
+    ): Unit = ensureInitialized().trackPlay(url, urlRef, videoMetadata, extraData)
 
-        /**
-         * Singleton instance factory Note: this must be called before [.sharedInstance]
-         *
-         * @param siteId        The Parsely public site id (eg "example.com")
-         * @param flushInterval The interval at which the events queue should flush, in seconds
-         * @param context             The current Android application context
-         * @param dryRun If set to `true`, events **won't** be sent to Parse.ly server
-         * @return The singleton instance
-         */
-        @JvmStatic
-        @JvmOverloads
-        public fun sharedInstance(
-            siteId: String,
-            flushInterval: Int = DEFAULT_FLUSH_INTERVAL_SECS,
-            context: Context,
-            dryRun: Boolean = false,
-        ): ParselyTracker {
-            return instance ?: run {
-                val newInstance = ParselyTrackerInternal(siteId, flushInterval, context, dryRun)
-                instance = newInstance
-                return newInstance
-            }
+    @JvmStatic
+    public fun trackPause(): Unit = ensureInitialized().trackPause()
+
+    @JvmStatic
+    public fun resetVideo(): Unit = ensureInitialized().resetVideo()
+
+    @JvmStatic
+    public fun flushEventQueue(): Unit = ensureInitialized().flushEventQueue()
+
+    @JvmStatic
+    public fun flushTimerIsActive(): Boolean = ensureInitialized().flushTimerIsActive()
+
+    /**
+     * Singleton instance factory Note: this must be called before [.sharedInstance]
+     *
+     * @param siteId        The Parsely public site id (eg "example.com")
+     * @param flushInterval The interval at which the events queue should flush, in seconds
+     * @param context             The current Android application context
+     * @param dryRun If set to `true`, events **won't** be sent to Parse.ly server
+     * @return The singleton instance
+     */
+    @JvmStatic
+    @JvmOverloads
+    public fun init(
+        siteId: String,
+        flushInterval: Int = DEFAULT_FLUSH_INTERVAL_SECS,
+        context: Context,
+        dryRun: Boolean = false,
+    ) {
+        if (instance != null) {
+            Logging.log("Parse.ly has been already initialized. Previous configuration will be overwritten.")
         }
+        instance = ParselyTrackerInternal(siteId, flushInterval, context, dryRun)
```

Comment:
I took an inspiration for this method from Sentry SDK ([link](https://github.com/getsentry/sentry-java/blob/2465853df157c4ad881166804873989c47c10f4d/sentry/src/main/java/io/sentry/Sentry.java#L218-L230)).

I think `ParselyTracker.init` is a clear name for the initialization method and it won't cause any doubts.

Citing [your commen](https://github.com/Parsely/parsely-android/pull/99#discussion_r1453748066)t @oguzkocer 

> A singleton that requires a configuration presents an interesting problem. What happens if the client tries to pass a different configuration after the instance is already initialized? (...) Assuming that we don't want to allow updating the configuration after the initialization, I think we can give them a heads up about this in the documentation and by printing a log.

I'm not entirely sure if we want or do not want to allow users to reinitialize the tracker. Previously we didn't allow, but maybe there's no reason to?

My only concern currently is: what happens if a user started the engagement session and then re-initialized SDK? I guess there might be a memory leak here, but haven't tested it _yet_.

Reply:
On the second thought, I've decided to disable re-initialization (35e9cdda5de5f63e1e5596bbcbf9d9b5c4626764). The SDK doesn't allow doing this now, and I see no good arguments to allow it.